### DISCUSSION TITLE
80286 support

### DISCUSSION THREAD

**Discussion by darvin** [dbalsom/marty#21] [Ideas]
How hard would be to extend emulator to support 80286?

**Comment by dbalsom** [comment]
I would like to do so eventually.  It will require some additional research to make it reasonably accurate. Unfortunately, the microcode for the 286 has not been decoded.

**Reply by darvin** [reply]
how everybody else overcomes it?

what about 386?

**Reply by bachimusprime** [reply]
@darvin They don't emulate the 286 cycle-accurately that's how lol

**Comment by dbalsom** [comment]
> how everybody else overcomes it?

Implementing the 286 instruction set is one thing, doing it in an accurate way is another. Does the world need a cycle-accurate 286? maybe not, but it is something that interests me for some reason.

**Reply by darvin** [reply]
isn't it's impossible to run cycle accurate 286 emulation - because the are no published microcodes?

what exactly codes in microprocessors do anyway? which ROM is code for microprocessor?

**Reply by dbalsom** [reply]
maybe not impossible but it makes it much more difficult.   microcode allows more complicated instructions to be built up from smaller one, each CPU instruction is essentially a small program that runs on the CPU. Sometimes this program is just a set of instructions that execute in order, which is simple to emulate, but sometimes there can be branches or conditions which makes it harder to determine what the program logic is.   

The microcode ROM is on the CPU die itself.  If you are interested in learning more about microcode this is a good blog:
[https://www.righto.com/](https://www.righto.com/)

**Comment by Torinde** [comment]
Related:
- https://github.com/640-KB/GLaBIOS/issues/41 - XT/286 computer
- https://github.com/andreas-jonsson/virtualxt/issues/72 - Orchid Tiny Turbo 286 accelerator card for XT
